### ISSUE TITLE
Can't use ScikitLearn

### ISSUE THREAD

**Issue by degravek** [cstjean/ScikitLearn.jl#6]
I can't for the life of me get "using ScikitLearn" to work in julia. This occurs after running Pkg.add("ScikitLearn") successfully. The error I get is:

ERROR: LoadError: LoadError: ArgumentError: Iterators not found in path
 in compilecache at loading.jl:393
 in require at /Applications/Julia-0.4.6.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include at /Applications/Julia-0.4.6.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include_from_node1 at /Applications/Julia-0.4.6.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include at /Applications/Julia-0.4.6.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include_from_node1 at /Applications/Julia-0.4.6.app/Contents/Resources/julia/lib/julia/sys.dylib
 [inlined code] from none:2
 in anonymous at no file:0
 in process_options at /Applications/Julia-0.4.6.app/Contents/Resources/julia/lib/julia/sys.dylib
 in _start at /Applications/Julia-0.4.6.app/Contents/Resources/julia/lib/julia/sys.dylib
while loading /Users/kyledegrave/.julia/v0.4/ScikitLearn/src/preprocessing.jl, in expression starting on line 3
while loading /Users/kyledegrave/.julia/v0.4/ScikitLearn/src/ScikitLearn.jl, in expression starting on line 73
ERROR: Failed to precompile ScikitLearn to /Users/kyledegrave/.julia/lib/v0.4/ScikitLearn.ji
 in error at /Applications/Julia-0.4.6.app/Contents/Resources/julia/lib/julia/sys.dylib
 in compilecache at loading.jl:400
 in require at /Applications/Julia-0.4.6.app/Contents/Resources/julia/lib/julia/sys.dylib

I've never had a problem with any other package, and I can't figure out why. Any help is definitely appreciated!

**Comment by cstjean** [comment]
Thank you for the bug report. ScikitLearn.jl requires Iterators.jl to be installed, but it was not specified in the REQUIRE file, so it wasn't installed automatically. You can fix it by running `Pkg.add("Iterators")`. I've updated REQUIRE, so it shouldn't be an issue for anyone else.

**Comment by cstjean** [comment]
Let me know if you run into other problems!

**Comment by degravek** [comment]
I am now able to do "using ScikitLearn", but when running the command @sk_import linear_model: LogisticRegression, I get the error:

ERROR: PyError (:PyImport_ImportModule) <type 'exceptions.ImportError'>
ImportError('cannot import name __check_build',)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sklearn/__init__.py", line 56, in <module>
    from . import __check_build

When I go to my directory /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sklearn/linear_model/, this is the contents:

__init__.py		logistic.pyc		ridge.pyc
__init__.pyc		omp.py			sag.py
base.py			omp.pyc			sag.pyc
base.pyc		passive_aggressive.py	sag_fast.so
bayes.py		passive_aggressive.pyc	setup.py
bayes.pyc		perceptron.py		setup.pyc
cd_fast.so		perceptron.pyc		sgd_fast.so
coordinate_descent.py	randomized_l1.py	stochastic_gradient.py
coordinate_descent.pyc	randomized_l1.pyc	stochastic_gradient.pyc
least_angle.py		ransac.py		tests
least_angle.pyc		ransac.pyc		theil_sen.py
logistic.py		ridge.py		theil_sen.pyc

It looks like I have no "LogisticRegression" there. However, if I use instead "logistic" or "bayes" in the command above, I get the same error message. I should say that I'm trying to follow the tutorial here: http://scikitlearnjl.readthedocs.io/en/latest/quickstart/

Thanks for your help!

**Comment by cstjean** [comment]
If you run Python and enter

from sklearn.linear_model import LogisticRegression


does it work? How did you install the Python version of scikit-learn, was
it done through Anaconda, or did ScikitLearn.jl perform the installation
for you?

**Comment by degravek** [comment]
If I enter from sklearn.linear_model import LogisticRegression, I get the error

Traceback (most recent call last):
  File "<pyshell#15>", line 1, in <module>
    from sklearn.linear_model import LogisticRegression
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sklearn/__init__.py", line 56, in <module>
    from . import __check_build
ImportError: cannot import name __check_build

And scikit-learn was installed using the command: pip install -U scikit-learn

**Comment by cstjean** [comment]
Then it seems that the error is with scikit-learn Python. I got a bunch of
results when googling the error message. The top suggestion is to pip
install scipy
<http://stackoverflow.com/questions/33193171/importerror-cannot-import-name-check-build-while-importing-tfidfvectorizer-fr>.
You could also uninstall scikit-learn and reinstall it via the anaconda
distribution <https://www.continuum.io/downloads>.

**Comment by degravek** [comment]
Okay, finally got ScikitLearn to work. I'm using a Mac, which automatically comes with Python pre-installed. I had also downloaded the newest version of Python from the Python website. I had tried running things like "pip install numpy", "pip install scipy", "pip install scikit-learn", etc., but nothing had allowed me to successfully run "using ScikitLearn" in julia. I tried installing miniconda, and ran things like "conda install scikit-learn", etc. but it did not work.

To fix things, I removed the additional version of Python I had installed from the Python site (not sure what effect this had), and I also removed miniconda. I then installed Anaconda, and everything was fine.

Long story short: just install Anaconda.

**Comment by cstjean** [comment]
Thank you for the follow-up. Yes, Python is especially messy on OSX because there can be multiple versions installed at the same time, and it's not clear which one PyCall uses. Glad it worked out!
